src/Renderer/ThreeExtended/PlanarControls.js

```diff
@@ -0,0 +1,746 @@
+    this.groundLevel = options.groundLevel || 200;
+
+    // min and max duration in seconds, for animated travels with 'auto' parameter
+    this.autoTravelTimeMin = options.autoTravelTimeMin || 1.5;
+    this.autoTravelTimeMax = options.autoTravelTimeMax || 4;
+
+    // max travel duration is reached for this travel distance
+    this.autoTravelTimeDist = options.autoTravelTimeDist || 20000;
+
+    // after a smartZoom, camera height above ground will be between these two values
+    this.smartZoomHeightMin = options.smartZoomHeightMin || 75;
+    this.smartZoomHeightMax = options.smartZoomHeightMax || 500;
+
+    // if set to true, animated travels have 0 duration
+    this.instantTravel = options.instantTravel || false;
+
+    this.minZenithAngle = options.minZenithAngle || 0 * Math.PI / 180;
+
+    // should be less than 90 deg (90 = parallel to the ground)
+    this.maxZenithAngle = (options.maxZenithAngle || 82.5) * Math.PI / 180;
+
+    // prevent the default contextmenu from appearing when right-clicking
+    // this allows to use right-click for input without the menu appearing
+    this.domElement.addEventListener('contextmenu', onContextMenu.bind(this), false);
+
+    // add this PlanarControl instance to the view's framerequesters
+    // with this, PlanarControl.update() will be called each frame
+    this.view.addFrameRequester(this);
+
+    this.state = STATE.NONE;
+    this.isCtrlDown = false;
+
+    // mouse movement
+    this.mousePosition = new THREE.Vector2();
+    this.lastMousePosition = new THREE.Vector2();
+    this.deltaMousePosition = new THREE.Vector2(0, 0);
+
+    // drag movement
+    this.dragStart = new THREE.Vector3();
+    this.dragEnd = new THREE.Vector3();
+    this.dragDelta = new THREE.Vector3();
+
+    // camera focus point : ground point at screen center
+    this.centerPoint = new THREE.Vector3(0, 0, 0);
+
+    // camera rotation
+    this.phi = 0.0;
+
+    // animated travel
+    this.travelEndPos = new THREE.Vector3();
+    this.travelStartPos = new THREE.Vector3();
+    this.travelStartRot = new THREE.Quaternion();
+    this.travelEndRot = new THREE.Quaternion();
+    this.travelAlpha = 0;
+    this.travelDuration = 0;
+    this.travelUseRotation = false;
+    this.travelUseSmooth = false;
+
+    // time management
+    this.deltaTime = 0;
+    this.lastElapsedTime = 0;
+    this.clock = new THREE.Clock();
+
+    // eventListeners handlers
+    this._handlerOnKeyDown = onKeyDown.bind(this);
+    this._handlerOnKeyUp = onKeyUp.bind(this);
+    this._handlerOnMouseDown = onMouseDown.bind(this);
+    this._handlerOnMouseUp = onMouseUp.bind(this);
+    this._handlerOnMouseMove = onMouseMove.bind(this);
+    this._handlerOnMouseWheel = onMouseWheel.bind(this);
+
+    /**
+    * PlanarControl update
+    * Updates the view and camera if needed, and handles the animated travel
+    */
+    this.update = function update() {
+        this.deltaTime = this.clock.getElapsedTime() - this.lastElapsedTime;
```

Comment:
Why don't you use the `dt` provided as an argument to your update function (see https://github.com/iTowns/itowns/blob/master/src/Core/MainLoop.js#L63)?